src/unlimited.cpp

```diff
@@ -1212,23 +1213,27 @@ bool CheckAndRequestExpeditedBlocks(CNode* pfrom)
     {
       BOOST_FOREACH(string& strAddr, mapMultiArgs["-expeditedblock"]) 
         {
-          // Add the peer's listening port if it is empty
-          int pos1 = strAddr.rfind(":");
-          int pos2 = strAddr.rfind("]:");
-          if (pos1 <= 0 && pos2 <= 0)
-              strAddr += ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
-
           string strListeningPeerIP;
           string strPeerIP = pfrom->addr.ToString();
-          pos1 = strPeerIP.rfind(":");
-          pos2 = strPeerIP.rfind("]:");
-          // Handle both ipv4 and ipv6 cases
-          if (pos1 <= 0 && pos2 <= 0) 
-              strListeningPeerIP = strPeerIP + ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
-          else if (pos1 > 0)
-              strListeningPeerIP = strPeerIP.substr(0, pos1) + ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
-          else
-              strListeningPeerIP = strPeerIP.substr(0, pos2) + ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
+          // Add the peer's listening port if it was provided (only misbehaving clients do not provide it)
+          if (pfrom->addrFromPort != 0)
+            {
+              int pos1 = strAddr.rfind(":");
+              int pos2 = strAddr.rfind("]:");
+              if (pos1 <= 0 && pos2 <= 0)
+                strAddr += ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
+
+              pos1 = strPeerIP.rfind(":");
+              pos2 = strPeerIP.rfind("]:");
+              // Handle both ipv4 and ipv6 cases
+              if (pos1 <= 0 && pos2 <= 0) 
+                strListeningPeerIP = strPeerIP + ':' + boost::lexical_cast<std::string>(pfrom->addrFromPort);
```

Comment:
I'll be honest here, I don't find mandatory curlies as a big deal. [Kevlin Henney](https://youtu.be/FyCYva9DhsI?t=1h7m1s) makes a case why it might be desirable to leave them off really for the sake of less lexical noise in the code.

Now I get why mandatory curlies might be desirable but I can't recall the last time this pitfall came back to bite me in the ass.